File: espritx-Symfony V4/espritx-main/src/Enum/GroupType.php
<?php

namespace App\Enum;

use App\Traits\RandomizableEnum;
use Elao\Enum\SimpleChoiceEnum;
use ReflectionClass;

/**
 * @extends ReadableEnum<string>
 *
 * @method static STUDENT
 * @method static SITE_STAFF
 * @method static FACULTY_STAFF
 * @method static TEACHERS
 * @method static SUPER_ADMIN
 */
final class GroupType extends SimpleChoiceEnum
{
  use RandomizableEnum;

  public const SUPER_ADMIN = 'super admin';
  public const STUDENT = 'student';
  public const SITE_STAFF = 'site staff';
  public const FACULTY_STAFF = 'faculty staff';
  public const TEACHERS = 'teachers';

  public function slufigy(): string
  {
    return strtolower(trim(preg_replace('/[^A-Za-z0-9-]+/', '_', $this->value)));
  }
}

File: EspritX_Java-Desktop/EspritX_Java-master/src/edu/esprit/enums/GroupType.java
package edu.esprit.enums;

import edu.esprit.lib.persistence.QueryableEnum;

public enum GroupType implements QueryableEnum<String> {
    SUPER_ADMIN("super admin"),
    STUDENT("student"),
    SITE_STAFF("site staff"),
    FACULTY_STAFF("faculty staff"),
    TEACHERS("teachers");

    private final String value;

    GroupType(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}
